### PR TITLE
PVNet vs PVLive Intraday Analysis report 

### PR DESCRIPTION
**Pull request by zakwatts**
A report created to showcase the difference in performance between PVNet and PVLive Intraday.

### CODE REVIEW

**Review by peterdudfield — CHANGES_REQUESTED** [review]
Just fix the formating on the `join_backtest.py` file.

File: reports/pvnet_comp.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+   * [Mean Absolute Error](#mean-absolute-error-1)
+   * [Mean Bias Error](#mean-bias-error-1)
+5. [Conclusion](#conclusion)
+6. [Appendix](#appendix)
+
+## Introduction
+
+In this report, we evaluate the performance of OCFs national PV forecast (PVNet) against the Shefield Solar PVLive Intraday forecast, using the updated PVLive forecast as the benchmark truth for this comparison. This report will look into various error metrics and discuss the implications of our findings. First we will compare how PVLive Intraday and PVNet perform against PVLive Updated over periods in which we have data for. Then a more comprehensive evaluation of PVNet vs PVLive Updated is completed across the year of 2022, including evaluating the performance across different forecast horizons.
+
+This report should be used to indicate the difference in the forecasting capaboilities between the two as well as provide a framework for future analysis to completed over an larger number of data points.
```
Comment (dfulu) [inline reports/pvnet_comp.md:25]: capaboilities -> capabilities

File: reports/pvnet_comp.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+---
+
+## Data Overview
+
+The aim of these results is to evaluate the performance of the OCF PVNet model against the PVLive Intraday forecast over 2022. As PVLive Intraday is forecast with just a single horizon (at t=0 (now)), an equivlent horizon has been extracted from the PVNet forecast to make it a fair comparison.
```
Comment (dfulu) [inline reports/pvnet_comp.md:31]: It is our 30-minute horizon forecast (at t0+30minutes), that you are comparing to PVLive. Maybe you meant to skip the unnecessary detail when saying "equivalent horizon", which is totally fair. Up to you if you want to be more explicit about what part of our forecast it comes from.

equivlent -> equivalent

File: reports/pvnet_comp.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+The number of datapoints that can be used to compare is restricted by two factors. Firstly, OCF started recording the data from PVLive intraday around halfway through 2022 (on 2022-06-14). Hence for the time preceeding this period we do not currently have the data available to compare against PVNet. Secondly, when creating the PVNet backtest there was some missing data, in paricular:
```
Comment (dfulu) [inline reports/pvnet_comp.md:33]: The last sentence,  I'd change this to something like "Secondly, some input data which the PVNet model relies on was missing from our historic archive. In particular:"

File: reports/pvnet_comp.md [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+* ~1 month of missing NWP data
+* ~1 month of missing satelitte data
+* 2 days every month in which the statelittes switch over.
+
+This missing data would not occur in a production environment and ultimately would misreprsent the performance of the model. Hence these forecasts have been ommited from the comparison. The quantities of data available can be seen in the chart below (fig 1) where the y axis represents the date, with the top indicating the start of 2022 and the bottom indicating the end of 2022 and the x axis labelling the different data used. The darkend areas illustrates data that is available where as the white areas show missing or ommitted data.
```
Comment (dfulu) [inline reports/pvnet_comp.md:39]: I might change this first two sentences to something like: "These inputs gaps do not occur in our production environment. These forecasts could not be made without the missing inputs, and hence have been omitted from the comparison"

File: reports/pvnet_comp.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+### Results Table
+
+| Forecast | MAE                     | RMSE                   | MBE              | R2
+|----------|-------------------------|------------------------|------------------|-------
+| PVLive Intraday | 197.37 ± 4.19    | 357.56 ± 7.33          | -187.91 ± 4.28   | 0.978
+| PVNet    | 129.75 ± 2.73           | 233.49 ± 5.21          | 6.30 ± 3.28      | 0.99
+
+The ± represents the standard error in the metric.
+
+PVNet has a statiscially significant MAE of 129.75 MW compared to PVLive Intradays MAE of 197.37 MW(shown to be statstically significant with a sample size of n = 5053). Across the data points tested PVNet has a 34.3% reduction in MAE compared to PVLive Intraday.
```
Comment (dfulu) [inline reports/pvnet_comp.md:59]: First sentence I'd maybe go "PVNet has a significantly better MAE of 129.75 MW compared to PVLive Intraday's MAE of 197.37 MW (..."

File: reports/pvnet_comp.md [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+| PVNet    | 194.14         | 370.62        | 233.43      |
+
+Across all metrics, PVNet has lower standard deviations compared to PVLive Intraday, sugesting that PVNets errors are more concentrated around the mean and as such are more reliable.
+
+### Mean Absolute Error (MAE)
+
+![Model Forecast vs. Actuals](./imgs/pvnet_vs_pvlin_MAE_monthly.png)
+*Fig 2: Average monthly MAE (MW) for PVNet and PVLive Intraday*
+
+Errors are larger in the summer months for both forecasts due to increased solar radiation which inturn leads to greater differenced between the forecasted and actual values. This error generally decreases as the years goes into the winter months. PVNet shows a significant improvement in MAE, espcially over June and July. This difference decreases before then a similar performance in MAE is observed in November.
```
Comment (dfulu) [inline reports/pvnet_comp.md:77]: Check few spelling/grammar things in these couple of sentences

File: reports/pvnet_comp.md [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+
+---
+
+## PVNet Model Evaluation
+
+Next this report will look at the performance of PVNet compared to PVLive updated across 2022. PVNet was evaluated against PVLive updated, where data was present for each, over the period of:
+
+* 2022-01-01 to 2022-11-19
+
+The results shown in the table below are from taking that 0th hour forecast (now), which is the same that was used for the PVLive Intraday comparison above, and comparing the results to PVLive Updated. This section also contains graphs which illustrate the performance of PVNet across its different forecasting horizons (0-8hours).
```
Comment (dfulu) [inline reports/pvnet_comp.md:153]: PVNet really predicts for the horizons (30mins- 8hours[480mins]). I think this slipped into the x-axis of some of your graphs as well

File: reports/pvnet_comp.md [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+The **Standard Deviations** of these metrics are calculated as follows:
+$$\sigma = \sqrt{\frac{1}{n} \sum_{i=1}^{n} \left( \text{Metric}_i - \bar{\text{Metric}} \right)^2}$$
+Where $\bar{\text{Metric}}$ is the mean of the metric across all data points.
+
+The **Standard Error of The Mean** of these metrics are calcuated as follows:
+$$ SE = \frac{\sigma}{\sqrt{n}}$$
+
+## Additional Graphs
+
+The following graphs show results for the root-mean-square deviation (RMSE).
```
Comment (dfulu) [inline reports/pvnet_comp.md:255]: In each of these graphs, I'd mention which figure in the main text they are similar to. Just to quickly provide context.

File: reports/pvnet_comp.md [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+
+![Error Distribution of Model A](./imgs/pvnet_vs_pvlin_RMSE_monthly.png)
+*Fig 21: Average monthly RMSE for PVNet and PVLive Intraday*
```
Comment (dfulu) [inline reports/pvnet_comp.md:258]: For example "Fig 21: (Similar to fig 2) Average monthly RMSE for PVNet and PVLive Intraday"

File: reports/pvnet_comp.md (hunk 7 above)
Comment (zakwatts) [reply reports/pvnet_comp.md]: Good spot, fixed

**Pull request closed without merging**